### ISSUE TITLE
Custom domain improvements

### ISSUE THREAD

**Issue by joshka** [github/docs#28333]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site

### What part(s) of the article would you like to see updated?

1. Consider adding a high level summary of the information that is necessary to setup a site that avoids only conveying the information step by step. Reference material like this is sometimes more useful than narrative. E.g.:

### Apex Domain
- Custom domain setting on `https://github.com/<organization/<repository>/settings/pages`:
  ```
  example.com
  ````
- A Records:
  ```plain
  185.199.108.153
  185.199.109.153
  185.199.110.153
  185.199.111.153
  ```
- AAAA Records:
  ```plain
  2606:50c0:8000::153
  2606:50c0:8001::153
  2606:50c0:8002::153
  2606:50c0:8003::153
  ```

2. The following seems inaccurate:
> Under "Custom domain", type your custom domain, then click Save. If you are publishing your site from a branch, this will create a commit that adds a CNAME file to the root of your source branch. If you are publishing your site with a custom GitHub Actions workflow , no CNAME file is created. For more information about your publishing source, see "[Configuring a publishing source for your GitHub Pages site](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)."

Given: a newly registered domain and an existing gh-pages site at org.github.io, with DNS not yet propagated,
Expected: no change to an existing site because the commit referenced would just be created as a PR, and no other change would be made.
Actual: Existing site immediately creates a redirect to the domain name, taking it offline.

**Comment by richardburce19** [comment]
.devcontainer

**Comment by stevecat** [comment]
Hi @joshka, thanks for opening this! I love the idea of adding a referential section to this article, such as a table listing the different DNS records and when you need to use them.

To make this change, and address your second point, I'd like to work closely with the GitHub Pages team to plan this work. For that reason, I've opened an issue internally and I'll close this one out. I've linked back to this issue and I'll follow up here when we've made some progress.

Thank you again for the suggestion, I think it'll be a massive improvement! ✨

**Comment by stevecat** [comment]
Hi again @joshka! I just wanted to drop a note here to say that we've been working on this and have just published a reference table of DNS record values, you can see that here:

https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#dns-records-for-your-custom-domain

Regarding your second point, it sounds like you expected the CNAME file to be created via pull request? That's not how the file is added, it's actually committed directly to the source branch, so will take effect immediately. While I was adding the table, I also made a small change here to try and clarify that:

```diff
- create a commit that adds a CNAME file to the root of your source branch
+ create a commit that adds a CNAME file directly to the root of your source branch
```

Do let me know if I've misunderstood, I'm happy to chat to our team and make further changes 😸 

Thanks again for opening this issue and suggesting this update!

**Comment by joshka** [comment]
>it sounds like you expected the CNAME file to be created via pull request? 

Yes. elsewhere in github, when changes are made on behalf of a user they are made by PR - e.g. adding a github action, editing a file directly on the site, adding a dependabot configuration. Making a direct modification like this is surprising behavior as it's out of pattern, especially for something that can cause an immediate availability change like this. This is likely less a doc problem and more a gh-pages problem though I'd guess.
